src/mame/taito/taito_f3_v.cpp

```diff
@@ -700,2153 +686,737 @@ u16 taito_f3_state::lineram_r(offs_t offset)
-			{
-				this_x += global_x;
-				this_y += global_y;
-			}
-			else /* Apply both scroll offsets */
-			{
-				this_x += global_x + subglobal_x;
-				this_y += global_y + subglobal_y;
-			}
-
-			y = this_y;
-			block_y = this_y;
-			x = this_x;
-			block_x = this_x;
-
-			block_zoom_x = spriteram16_ptr[current_offs + 0 + 1];
-			block_zoom_y = spriteram16_ptr[current_offs + 0 + 1] >> 8;
-
-			x_addition_left = 8;
-			calc_zoom(x_addition, x_addition_left, block_zoom_x);
-
-			y_addition_left = 8;
-			calc_zoom(y_addition, y_addition_left, block_zoom_y);
+		if (spr[7] != 0) {
+			logerror("unknown sprite word 7: %4x\n", spr[7]);
 		}
 
-		/* These features are common to sprite and block parts */
-		flipx = spritecont & 0x1;
-		flipy = spritecont & 0x2;
-		multi = spritecont & 0x8;
-		last_x = x;
-		last_y = y;
+		// Check if the sprite list jump bit is set
+		// we have to check this AFTER processing sprite commands because recalh uses a sprite command and jump in the same sprite
+		// i wonder if this should go after other sprite processsing as well?  can a regular sprite have a jump ?
+		if (BIT(spr[6], 15)) {
+			const int new_offs = BIT(spr[6], 0, 10);
+			if (new_offs == offs) // could this be ≤ ? -- NO! RECALH USES BACKWARDS JUMPS!!
+				break; // optimization, edge cases to watch for: looped sprite block commands?
 
-		if (!sprite) continue;
-		if (!x_addition || !y_addition) continue;
-
-		if (m_flipscreen)
-		{
-			const int tx = 512 - x_addition - x;
-			const int ty = 256 - y_addition - y;
-
-			if (tx + x_addition <= min_x || tx > max_x || ty + y_addition <= min_y || ty > max_y) continue;
-			sprite_ptr->x = tx;
-			sprite_ptr->y = ty;
-			sprite_ptr->flipx = !flipx;
-			sprite_ptr->flipy = !flipy;
-		}
-		else
-		{
-			if (x + x_addition <= min_x || x > max_x || y + y_addition <= min_y || y > max_y) continue;
-			sprite_ptr->x = x;
-			sprite_ptr->y = y;
-			sprite_ptr->flipx = flipx;
-			sprite_ptr->flipy = flipy;
+			offs = new_offs - 1; // subtract because we increment in the for loop
 		}
 
-		sprite_ptr->code = sprite;
+		const u8 spritecont = spr[4] >> 8;
+		const bool lock = BIT(spritecont, 2);
+		if (!lock)
+			color = spr[4] & 0xFF;
+		const u8 scroll_mode = BIT(spr[2], 12, 4);
+		const u16 zooms = spr[1];
+		x.update(scroll_mode, spr[2] & 0xFFF, multi, BIT(spritecont, 4+2, 2), zooms & 0xFF);
+		y.update(scroll_mode, spr[3] & 0xFFF, multi, BIT(spritecont, 4+0, 2), zooms >> 8);
+		multi = BIT(spritecont, 3);
+
+		const int tile = spr[0] | (BIT(spr[5], 0) << 16);
+		if (!tile) continue; // todo: is this the correct way to tell if a sprite exists?
+
+		const fixed8 tx = m_flipscreen ? (512<<8) - x.block_scale*16 - x.pos : x.pos;
+		const fixed8 ty = m_flipscreen ? (256<<8) - y.block_scale*16 - y.pos : y.pos;
+
+		if (tx + x.block_scale*16 <= visarea.min_x<<8 || tx > visarea.max_x<<8 || ty + y.block_scale*16 <= visarea.min_y<<8 || ty > visarea.max_y<<8)
+			continue;
```

Comment:
You now increment `total_sprites` *before* checks that cause the sprite to be ignored, while previously it happened *after* those checks.  This changes the way the sprite limit works – previously it excluded ignored sprites, now it includes them.  Is that intentional?

Reply:
`total_sprites` is a safeguard to prevent infinite loops; under normal circumstances the limit is never reached

Reply:
> `total_sprites` is a safeguard to prevent infinite loops; under normal circumstances the limit is never reached

Yes, but it will be reached faster with this change than it would have been before.

Reply:
The limit is equal to the number of entries in sprite ram (1024), so the **only** way to reach that limit is by reading the same sprite multiple times (i.e. the sprite list has a jump command which jumps backwards and creates an infinite loop)
(exiting a little sooner is fine, since the duplicate sprites would just get drawn on top of each other anyway)

If there's an infinite loop containing only nonvisible sprites, the old code gets stuck forever because total_sprites never increments